### DISCUSSION TITLE
What's the dbmate schema name for Clickhouse?

### DISCUSSION THREAD

**Discussion by detj** [amacneil/dbmate#506] [Support]
**Does dbmate automatically create its own schema named `dbmate` for Clickhouse to store `schema_migrations`?**

Unlike, postgres, I observed that dbmate did not create a `dbmate` schema to store `dbmate.schema_migrations`. Instead it is storing migrations in `default.schema_migrations`. **Is this expected behavior?**

Here's the Clickhouse query I ran to verify it.

```
4ed1e7564cf0 :) select name from system.databases;

SELECT name
FROM system.databases

Query id: 01573439-744e-4001-bbb4-93c45aeb3e8f

┌─name───────────────┐
│ INFORMATION_SCHEMA │
│ default            │
│ information_schema │
│ system             │
└────────────────────┘

4 rows in set. Elapsed: 0.003 sec.
```

```
4ed1e7564cf0 :) select * from default.schema_migrations;

SELECT *
FROM default.schema_migrations

Query id: 8e441714-f290-45b9-90af-9741011cd0f0

┌─version────────┬──────────────────ts─┬─applied─┐
│ 20231117020810 │ 2023-12-13 06:23:31 │       1 │
└────────────────┴─────────────────────┴─────────┘

1 row in set. Elapsed: 0.005 sec.
```

**here's how my `.env` looks like**

```sh
DATABASE_URL=clickhouse://default:@127.0.0.1:9000/default
DBMATE_MIGRATIONS_DIR=.
DBMATE_SCHEMA_FILE=./schema.sql
DBMATE_WAIT=true
```

**Comment by dossy** [comment]
> **Does dbmate automatically create its own schema named `dbmate` for Clickhouse to store `schema_migrations`?**

No.

> Unlike, postgres, I observed that dbmate did not create a `dbmate` schema to store `dbmate.schema_migrations`. Instead it is storing migrations in `default.schema_migrations`. **Is this expected behavior?**

Yes.

By default, dbmate creates the migrations table determined by either the command line argument `--migrations-table value` or the environment variable `$DBMATE_MIGRATIONS_TABLE` or the default value `schema_migrations`, which will be created in the database you specify in the `DATABASE_URL` -- in your case, `default`.

If you'd like to use a different migrations table in a separate database, you will need to ensure that database exists already, then specify it as a fully qualified table name to dbmate with either the `--migrations-table value` command line argument or the environment variable `$DBMATE_MIGRATIONS_TABLE`.

For example, if you want to store the migrations table in the database named `hello` and in a table named `world`, you would either use `--migrations-table hello.world` or set the environment variable `DBMATE_MIGRATIONS_TABLE=hello.world` before invoking dbmate.
